src/main/java/jenkins/plugins/slack/SlackNotifier.java

```diff
@@ -325,4 +329,186 @@ public FormValidation doTestConnection(@QueryParameter("slackTeamDomain") final
+            this.notifyUnstable = notifyUnstable;
+            this.notifyBackToNormal = notifyBackToNormal;
+            this.notifyRepeatedFailure = notifyRepeatedFailure;
+            this.includeTestSummary = includeTestSummary;
+            this.showCommitList = showCommitList;
+            this.includeCustomMessage = includeCustomMessage;
+            this.customMessage = customMessage;
+        }
+
+        @Exported
+        public String getTeamDomain() {
+            return teamDomain;
+        }
+
+        @Exported
+        public String getToken() {
+            return token;
+        }
+
+        @Exported
+        public String getRoom() {
+            return room;
+        }
+
+        @Exported
+        public boolean getStartNotification() {
+            return startNotification;
+        }
+
+        @Exported
+        public boolean getNotifySuccess() {
+            return notifySuccess;
+        }
+
+        @Exported
+        public boolean getShowCommitList() {
+            return showCommitList;
+        }
+
+        @Override
+        public boolean prebuild(AbstractBuild<?, ?> build, BuildListener listener) {
+            return super.prebuild(build, listener);
+        }
+
+        @Exported
+        public boolean getNotifyAborted() {
+            return notifyAborted;
+        }
+
+        @Exported
+        public boolean getNotifyFailure() {
+            return notifyFailure;
+        }
+
+        @Exported
+        public boolean getNotifyNotBuilt() {
+            return notifyNotBuilt;
+        }
+
+        @Exported
+        public boolean getNotifyUnstable() {
+            return notifyUnstable;
+        }
+
+        @Exported
+        public boolean getNotifyBackToNormal() {
+            return notifyBackToNormal;
+        }
+
+        @Exported
+        public boolean includeTestSummary() {
+            return includeTestSummary;
+        }
+
+        @Exported
+        public boolean getNotifyRepeatedFailure() {
+            return notifyRepeatedFailure;
+        }
+
+        @Exported
+        public boolean includeCustomMessage() {
+            return includeCustomMessage;
+        }
+
+        @Exported
+        public String getCustomMessage() {
+            return customMessage;
+        }
+
+    }
+
+    @Extension public static final class Migrator extends ItemListener {
+        @SuppressWarnings("deprecation")
+        @Override
+        public void onLoaded() {
+            logger.info("Starting Settings Migration Process");
+            for (AbstractProject<?, ?> p : Jenkins.getInstance().getAllItems(AbstractProject.class)) {
+                logger.info("processing Job: " + p.getName());
+
+                final SlackJobProperty slackJobProperty = p.getProperty(SlackJobProperty.class);
+
+                if(slackJobProperty == null) {
+                    logger.info("Configuration is already uptodate, exiting migration");
+                    return;
```

Comment:
I tested this PR, but I had some Problems. Because I had jobs that were created before the plugin was installed, the didn't have an old `slackJobProperty`. You should use `continue` instead of `return`.